pypeit/spectrographs/keck_deimos.py

```diff
@@ -1256,8 +1430,11 @@ def list_detectors(self):
             the array is 2D, there are detectors separated along the dispersion
             axis.
         """
-        return np.array([detector_container.DetectorContainer.get_name(i+1) 
-                            for i in range(self.ndet)]).reshape(2,-1)
+        if mosaic:
+            return np.array(['MSC{:02d}'.format(i+1) for i in range(len(self.allowed_mosaics))])
```

Comment:
I would use the `pypeit.images.mosaic.Mosaic.get_name` method to set the mosaic name.